pkg/cluster-runtime/installer.go

```diff
@@ -118,71 +124,126 @@ func getWorkerIPList(infraDriver infradriver.InfraDriver) []net.IP {
 	return workers
 }
 
-func (i *Installer) Install() (registry.Driver, runtime.Driver, error) {
-	master0 := i.infraDriver.GetHostIPListByRole(common.MASTER)[0]
-	masters := i.infraDriver.GetHostIPListByRole(common.MASTER)
-	workers := getWorkerIPList(i.infraDriver)
-	all := append(masters, workers...)
+func (i *Installer) Install() error {
+	var (
+		masters    = i.infraDriver.GetHostIPListByRole(common.MASTER)
+		master0    = masters[0]
```

Comment:
what if the length of masters is equal with zero?

Reply:
the length of masters will not equal to zero during the installation process, that's because we set default master role  in the beginning which was got from user input.